src/main/java/jarvis/logic/commands/AddConsultCommand.java

```diff
@@ -0,0 +1,98 @@
+            + PREFIX_START_DATE_TIME + "START_DATE_TIME "
+            + PREFIX_END_DATE_TIME + "END_DATE_TIME "
+            + PREFIX_STUDENT_INDEX + "STUDENT_INDEX...\n"
+            + "Example: " + COMMAND_WORD + " "
+            + PREFIX_LESSON + "Consultation on recursion "
+            + PREFIX_START_DATE_TIME + "2022-10-14T20:00 "
+            + PREFIX_END_DATE_TIME + "2022-10-14T21:00 "
+            + PREFIX_STUDENT_INDEX + "3 "
+            + PREFIX_STUDENT_INDEX + "4";
+
+    public static final String MESSAGE_SUCCESS = "New consultation added: %1$s";
+    public static final String MESSAGE_DUPLICATE_CONSULT = "This consultation already exists in JARVIS";
+
+    private final LessonDesc consultDesc;
+    private final TimePeriod consultPeriod;
+    private final Set<Index> studentIndexSet;
+
+    /**
+     * Creates an AddConsultCommand to add the specified {@code Consult} with
+     * {@code consultDesc, consultPeriod, studentIndexSet}
+     */
+    public AddConsultCommand(LessonDesc consultDesc, TimePeriod consultPeriod,
+                                  Set<Index> studentIndexSet) {
+        requireNonNull(consultDesc);
+        requireNonNull(consultPeriod);
+        requireNonNull(studentIndexSet);
+        this.consultDesc = consultDesc;
+        this.consultPeriod = consultPeriod;
+        this.studentIndexSet = studentIndexSet;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        List<Student> lastShownList = model.getFilteredStudentList();
+
+        Set<Student> studentSet = new HashSet<>();
+        for (Index studentIndex : studentIndexSet) {
+            if (studentIndex.getZeroBased() >= lastShownList.size()) {
+                throw new CommandException(Messages.MESSAGE_INVALID_STUDENT_DISPLAYED_INDEX);
+            }
+            Student studentToAdd = lastShownList.get(studentIndex.getZeroBased());
+            studentSet.add(studentToAdd);
+        }
+        LessonAttendance consultAttendance = new LessonAttendance(studentSet);
+        LessonNotes consultNotes = new LessonNotes(studentSet);
+        Consult consultToAdd = new Consult(consultDesc, consultPeriod, consultAttendance, consultNotes);
+
+        if (model.hasLesson(consultToAdd)) {
+            throw new CommandException(MESSAGE_DUPLICATE_CONSULT);
+        }
+
+        model.addLesson(consultToAdd);
+        return new CommandResult(String.format(MESSAGE_SUCCESS, consultToAdd), true, false);
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        return other == this // short circuit if same object
+                || (other instanceof AddConsultCommand // instanceof handles nulls
+                && consultDesc.equals(((AddConsultCommand) other).consultDesc))
+                && consultPeriod.equals(((AddConsultCommand) other).consultPeriod)
+                && studentIndexSet.equals(((AddConsultCommand) other).studentIndexSet);
+    }
+}
```

Comment:
I think we can just typecast the object one time and assign it to a new variable. Same for Mastery Check and Studio.